server/canvas_api_response.rb

```diff
@@ -0,0 +1,93 @@
+###### CTools direct URL provider ################
+# Format the CTools direct response into json format that UI expects.
+
+require_relative './Logging'
+require 'rest-client'
+
+# Handle interpretation and reformatting of a CTools direct call.
+# The incoming data will NOT have the WAPI wrapper.
+
+## Output format is defined in: https://docs.google.com/document/d/1KfhK6aMW1FdZdQwzTj5UO_zHq1qG6Fnpq80yWFV8hN4/edit
+
+class CanvasAPIResponse
+
+  # Store the body of the response.  All output from Canvas API goes through this class.
+  attr_accessor :body_string, :body_json
+
+  ## TODO: allow input as string or json
+
+  # Input is in string format
+  def initialize(body)
+    @body_string = body
+    @body_json = JSON.parse(body)
+    @canvas_instance_prefix = 'https://umich.test.instructure.com.dummy'
```

Comment:
Good point. It is used for the moment, I've filed a Jira to find and use
the real value.

On Tue, Oct 20, 2015 at 10:39 AM, Lance E Sloan <notifications@github.com>
wrote:

> In server/canvas_api_response.rb
> <https://github.com/tl-its-umich-edu/StudentDashboard/pull/202#discussion_r42503857>
> :
>
> > +# The incoming data will NOT have the WAPI wrapper.
> > +
> > +## Output format is defined in: https://docs.google.com/document/d/1KfhK6aMW1FdZdQwzTj5UO_zHq1qG6Fnpq80yWFV8hN4/edit
> > +
> > +class CanvasAPIResponse
> > +
> > +  # Store the body of the response.  All output from Canvas API goes through this class.
> > +  attr_accessor :body_string, :body_json
> > +
> > +  ## TODO: allow input as string or json
> > +
> > +  # Input is in string format
> > +  def initialize(body)
> > +    @body_string = body
> > +    @body_json = JSON.parse(body)
> > +    @canvas_instance_prefix = 'https://umich.test.instructure.com.dummy'
>
> Is this the final value to be used in production?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tl-its-umich-edu/StudentDashboard/pull/202/files#r42503857>
> .
>